controller/DeliveryServer.php

```diff
@@ -51,16 +52,33 @@ class DeliveryServer extends ProctoringDeliveryServer
      */
     public function awaitingAuthorization()
     {
+        $deliveryExecution = $this->getCurrentDeliveryExecution();
+        $this->validateAutoStart($deliveryExecution);
+
         try {
             parent::awaitingAuthorization();
 
         }catch (QtiTestExtractionFailedException $e) {
             throw new LtiException($e->getMessage());
         }
-        $deliveryExecution = $this->getCurrentDeliveryExecution();
+
         $this->setData('cancelUrl', _url('cancelExecution', 'DeliveryServer', 'ltiProctoring', ['deliveryExecution' => $deliveryExecution->getIdentifier()]));
     }
 
+    private function validateAutoStart(DeliveryExecution $deliveryExecution): void
+    {
+        $user = common_session_SessionManager::getSession()->getUser();
+        $redirectUrl = $this->getAutoStartProctorService()->execute($deliveryExecution, $user, $this->getSession());
+        if ($redirectUrl) {
+            $this->redirect($redirectUrl);
+        }
+    }
+
+    private function getAutoStartProctorService(): AutoStartProctorService
+    {
+        return $this->getServiceManager()->getContainer()->get(AutoStartProctorService::class);
```

Comment:
`getServiceManager()` is deprecated.
Please use the following container accessor instead.

```suggestion
        return $this->getPsrContainer()→get(AutoStartProctorService::class);
```